ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -324,21 +324,23 @@ type FSharpOverridingMembersBuilder() =
         use disableFormatter = new DisableCodeFormatter()
 
         let typeDecl = context.Root :?> IFSharpTypeDeclaration
-        let typeRepr = typeDecl.TypeRepresentation
 
-        match typeRepr with
+        match typeDecl.TypeRepresentation with
         | :? IUnionRepresentation as unionRepr ->
             unionRepr.UnionCasesEnumerable
             |> Seq.tryHead
             |> Option.iter EnumCaseLikeDeclarationUtil.addBarIfNeeded
-        | :? ITypeAbbreviationRepresentation as abbrRepr ->
-            if isNotNull abbrRepr.FirstChild then
-                addNodesBefore abbrRepr.FirstChild [
-                    FSharpTokenType.BAR.CreateLeafElement()
-                    Whitespace()
-                ] |> ignore
+        | :? ITypeAbbreviationRepresentation as abbrRepr when abbrRepr.CanBeUnionCase ->
+            let factory = typeDecl.CreateElementFactory()
+            let declGroup = factory.CreateModuleMember($"type U = | {abbrRepr.AbbreviatedTypeOrUnionCase.DeclaredName}")
```

Comment:
Please escape the name if needed (there should be some examples in the factory code, try to search for `mangle` or `escape`). This can be a test case:
```fsharp
type U = ``A B``
```

Also it's usually safer to use `SourceName` instead of `DeclaredName`, as the latter one is what we provide to R#, and it needs the compiled name, which could be different from the source name in F#.

Reply:
Done.